test/models/attachment_test.rb

```diff
@@ -106,6 +106,18 @@ class AttachmentTest < ActiveSupport::TestCase
       assert attachment.path.end_with?(expected_path)
     end
 
+    test "set a attachment to nil" do
+      blob = create_blob(data: "NewData")
+
+      @user.avatar.attach(blob, filename: "test.txt")
+
+      @user.update(avatar: nil)
+
+      assert_not_predicate @user.avatar, :present?
+      assert_equal 0, blob.attachments_count
```

Comment:
Deze count wordt via triggers bijgewerkt dus Rails ziet hier niks van. Ik vermoed dat deze count gedurende de hele spec 0 blijft omdat alleen het object in memory getest wordt.

Reply:
Goed punt!